crates/astria-core/src/execution/v2/mod.rs

```diff
@@ -0,0 +1,699 @@
+}
+
+impl ExecutionSession {
+    #[must_use]
+    pub fn session_id(&self) -> &String {
+        &self.session_id
+    }
+
+    #[must_use]
+    pub fn execution_session_parameters(&self) -> &ExecutionSessionParameters {
+        &self.execution_session_parameters
+    }
+
+    #[must_use]
+    pub fn commitment_state(&self) -> &CommitmentState {
+        &self.commitment_state
+    }
+}
+
+impl Protobuf for ExecutionSession {
+    type Error = ExecutionSessionError;
+    type Raw = raw::ExecutionSession;
+
+    fn try_from_raw_ref(raw: &Self::Raw) -> Result<Self, Self::Error> {
+        let raw::ExecutionSession {
+            session_id,
+            execution_session_parameters,
+            commitment_state,
+        } = raw;
+        let execution_session_parameters = execution_session_parameters
+            .as_ref()
+            .ok_or_else(Self::Error::missing_execution_session_parameters)?;
+        let execution_session_parameters =
+            ExecutionSessionParameters::try_from_raw_ref(execution_session_parameters)
+                .map_err(Self::Error::execution_session_parameters)?;
+        let commitment_state = commitment_state
+            .as_ref()
+            .ok_or_else(Self::Error::missing_commitment_state)?;
+        let commitment_state = CommitmentState::try_from_raw_ref(commitment_state)
+            .map_err(Self::Error::commitment_state)?;
+        Ok(Self {
+            session_id: session_id.clone(),
+            execution_session_parameters,
+            commitment_state,
+        })
+    }
+
+    fn to_raw(&self) -> Self::Raw {
+        let Self {
+            session_id,
+            execution_session_parameters,
+            commitment_state,
+        } = self;
+        let execution_session_parameters = execution_session_parameters.to_raw();
+        let commitment_state = commitment_state.to_raw();
+        Self::Raw {
+            session_id: session_id.clone(),
+            execution_session_parameters: Some(execution_session_parameters),
+            commitment_state: Some(commitment_state),
+        }
+    }
+}
+
+// An error when transforming a [`raw::ExecutionSessionParameters`] into a
+// [`ExecutionSessionParameters`].
+#[derive(Debug, thiserror::Error)]
+#[error(transparent)]
+pub struct ExecutionSessionParametersError(ExecutionSessionParametersErrorKind);
+
+impl ExecutionSessionParametersError {
+    fn incorrect_rollup_id_length(inner: IncorrectRollupIdLength) -> Self {
+        Self(ExecutionSessionParametersErrorKind::IncorrectRollupIdLength(inner))
+    }
+
+    fn no_rollup_id() -> Self {
+        Self(ExecutionSessionParametersErrorKind::NoRollupId)
+    }
+
+    fn invalid_sequencer_start_block_height(inner: tendermint::Error) -> Self {
+        Self(ExecutionSessionParametersErrorKind::InvalidSequencerStartBlockHeight(inner))
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+enum ExecutionSessionParametersErrorKind {
+    #[error("`rollup_id` field contained an invalid rollup ID")]
```

Comment:
Make it explicit that this is a field:
```suggestion
    #[error("field `.rollup_id` field contained an invalid rollup ID")]
```

Similary for the other variants below.

Reply:
80238e7